third_party/move/move-binary-format/src/file_format_common.rs

```diff
@@ -405,9 +405,14 @@ pub const VERSION_5: u32 = 5;
 ///  + u16, u32, u256 integers and corresponding Ld, Cast bytecodes
 pub const VERSION_6: u32 = 6;
 
-// Mark which version is the latest version
+/// Mark which version is the latest version
 pub const VERSION_MAX: u32 = VERSION_6;
 
+/// A unique version value which is used for experimental code which is not allowed in
+/// production. The bytecode deserializer accepts modules with this version only when the
+/// cargo feature `testing` is enabled.
+pub const VERSION_EXPERIMENTAL: u32 = u32::MAX;
```

Comment:
Is there supposed to be a different version of the bytecode for the V2 compiler, or will it be completely compatible with the old version?

Reply:
In the first iteration, we aim at v6 (the current version), so fully compatible.